velox/exec/HashProbe.cpp

```diff
@@ -1219,6 +1219,15 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
         outputTableRows_[numPassed] = outputTableRows_[i];
         rawOutputProbeRowMapping[numPassed++] = rawOutputProbeRowMapping[i];
       }
+      // Last row in the sub-batch which is not added to outputTableRows_ but
+      // has filter_ as false needs to be added with NULLs as this could
+      // increment the value of numPassed by 1 with the next sub-batch
```

Comment:
Instead, of 

> as this could increment the value of numPassed by 1 with the next sub-batch  processed.

I would say something like this:
as the current sub-batch rows need to be accounted for before the next sub-batch begins.

Reply:
Done.